src/rvsdg/tree_unique/to_tree.rs

```diff
@@ -55,37 +61,39 @@ impl<'a> RegionTranslator<'a> {
         res
     }
 
-    /// Build a translator and translate
-    /// the operands to the tree encoding.
-    /// Produces a tree-encoded term that evaluates
-    /// to a tuple containing results.
-    fn translate(num_args: usize, nodes: &'a Vec<RvsdgBody>, results: Vec<Operand>) -> Expr {
-        let mut translator = RegionTranslator {
+    /// Make a new translator for a region with
+    /// num_args and the given nodes.
+    fn new(num_args: usize, nodes: &'a Vec<RvsdgBody>) -> RegionTranslator {
+        RegionTranslator {
             num_args,
             bindings: Vec::new(),
             index_of: HashMap::new(),
             nodes,
-        };
-
-        let mut result_indices = Vec::new();
-        for result in results {
-            result_indices.push(translator.translate_operand(result));
         }
+    }
 
-        let mut expr = sequence(result_indices.iter().map(|i| get(arg(), *i)).collect());
+    /// Wrap the given expression in all the
+    /// bindings that have been generated.
+    fn build_translation(&self, inner: Expr) -> Expr {
+        let mut expr = inner;
 
-        for binding in translator.bindings.into_iter().rev() {
-            expr = cbind(binding, expr);
+        for binding in self.bindings.iter().rev() {
+            expr = cbind(binding.clone(), expr);
         }
         expr
     }
 
-    fn translate_operand(&mut self, operand: Operand) -> usize {
+    /// Returns a pure expression (e.g. `getarg(0)`) that
```

Comment:
I don't understand the e.g. What is getarg?

Reply:
`getarg` is a helper that desugars to `(Get (Arg (Id 0)) 0)`